lib/pbench/server/api/resources/users_api.py

```diff
@@ -282,23 +297,52 @@ def post(self):
                         "message": "failure message"
                     }
         """
-        auth_token = self.auth.get_auth_token(self.logger)
-        user = Auth.verify_auth(auth_token=auth_token)
+        auth_token = Auth.get_auth_token()
+        state = Auth.verify_token_only(auth_token)
 
-        # "None" user represents that either the token is not present in our database or it is an expired token.
-        # Expired token is already deleted by now if we reach here.
-        # In either case we will return suceess to a client, since the same token can not be used again.
+        # The only time a token won't be in the database is if the token is
+        # not a valid token.  Valid tokens that are not expired should be in
+        # the database, and expired tokens might or might not be in the
+        # database.  If the valid token is found in the database associated
+        # with user, remove it.
 
-        if user:
-            try:
-                ActiveTokens.delete(auth_token)
-            except Exception:
-                self.logger.exception("Exception occurred deleting an auth token")
-                abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
-            else:
-                self.logger.debug("User {} logged out", user.username)
+        if state == Auth.TokenState.INVALID:
+            current_app.logger.info("User logout with invalid token: {}", auth_token)
         else:
-            self.logger.info("User logout with invalid or expired token")
+            token = ActiveToken.query(auth_token)
+            if token:
+                user = token.user
+                if not user:
+                    current_app.logger.info(
+                        "Unknown user logged out with verified or expired token"
+                    )
```

Comment:
I'd rather not spend too much time on this given we are moving to KeyCloak.  There is little to be gained by changing this PR further in this area.

Reply:
I'm confused:  this PR is making a bunch of change in this area, but you say there is little to be gained from it because of the impending move to KeyCloak...so, why are we considering these changes, then?  Should we drop this change from the PR and get it down to just the stuff we need in order to get to KeyCloak.

(However, even with KeyCloak, I think we're going to need "API keys" or "robot tokens" or whatever they are called...aren't we going to want all this stuff in order to provide them??)

Reply:
The previous code used `verify_auth()` token, which means that every logout API call would result in a database operation lookup whether or not the token was valid.  That is not a good place to be for an API.

Then, the previous code would only delete the token if that token was associated with a user.

The current code now does:

 * NOTHING if the token is invalid
 * Deletes the token for the user
 * Maintains the same internal error return behavior
 * Improves the logging for the various cases

So it is a refactoring to improve the logging and expose internal errors.

Reply:
I think your refactoring is a good thing, and I'd like to see it carried all the way through.

Thanks to your changes, we now query the database only if the token is valid, which is a good improvement.  However, if the query returns a token, then I think we should delete it on logout...and, it shouldn't matter whether `token.user` is non-null.  Ideally, we would confirm that the token matched the current user, but I don't think we can do that if `token.user` is null.

In this context, deleting the token from the database is the conservative approach.  The only value I see to testing `token.user` is to make it easier to log the resulting value later.  However, I think that that is easily handled without creating a separate path through the function (just set up a local variable with the string to be used for the username -- we appear to use it in all the subsequent paths).

Given that you have other changes to make to this PR, I think it's worth making this change, as well.